### PR TITLE
Add StatefulSet resource

### PR DESCRIPTION
**Pull request by tremes**

### CODE REVIEW

File: test/statefulset-test.js [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+      .matchHeader('authorization', `Bearer ${clientConfig.user.token}`) // taken from the config
+      .get(`/oapi/v1/namespaces/${clientConfig.context.namespace}/statefulsets/${statefulSetName}`)
+      .reply(200, {kind: 'StatefulSet'});
+
+    const findResult = client.statefulsets.find(statefulSetName).then((statefulset) => {
+      t.equal(statefulset.kind, 'StatefulSet', 'returns an object with StatefulSet');
+      t.end();
+    }).catch(e => {
+      console.log(e);
+    });
```
Comment (tremes) [inline test/statefulset-test.js]: I can remove this catch once the tests will be passing

File: test/statefulset-test.js [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+test('find - statefulsets - basic find', (t) => {
+  openshiftRestClient(settings).then((client) => {
+    t.equal(typeof client.statefulsets.find, 'function', 'There is a find method on the statefulsets object');
+
+    const clientConfig = privates.get(client).config;
+    const statefulSetName = 'cool-statefulset-name-1';
+
+    nock(clientConfig.cluster)
+      .matchHeader('authorization', `Bearer ${clientConfig.user.token}`) // taken from the config
+      .get(`/oapi/v1/namespaces/${clientConfig.context.namespace}/statefulsets/${statefulSetName}`)
```
Comment (lholmquist) [inline test/statefulset-test.js]: This should be `/apis/apps/v1beta1` instead of `/oapi/v1/`

File: test/statefulset-test.js [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+    t.equal(typeof client.statefulsets.create, 'function', 'There is a create method on the statefulsets object');
+
+    const clientConfig = privates.get(client).config;
+    const statefulset = {
+      kind: 'StatefulSet'
+    };
+
+    nock(clientConfig.cluster)
+      .matchHeader('authorization', `Bearer ${clientConfig.user.token}`) // taken from the config
+      .post(`/oapi/v1/namespaces/${clientConfig.context.namespace}/statefulsets`)
```
Comment (lholmquist) [inline test/statefulset-test.js]: This should be `/apis/apps/v1beta1` instead of `/oapi/v1/`

File: test/statefulset-test.js [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+    });
+
+    t.equal(findResult instanceof Promise, true, 'should return a Promise');
+  });
+});
+
+test('find - statefulsets - find - no statefulset name', (t) => {
+  openshiftRestClient(settings).then((client) => {
+    client.statefulsets.find().catch((err) => {
+      t.equal(err.message, 'StatefulSet Name is required', 'error message should return');
```
Comment (lholmquist) [inline test/statefulset-test.js]: Tests fail because it is looking for `Stateful Set`

File: test/statefulset-test.js (hunk 2 above)
Comment (tremes) [reply test/statefulset-test.js]: Oh yes...stupid mistake:). Thanks

File: test/statefulset-test.js (hunk 4 above)
Comment (tremes) [reply test/statefulset-test.js]: Again...stupid mistake:) Thanks a lot for review. I am about to push an update.

**Comment by lholmquist** [comment]
@tremes can you give this a Rebase,  i've merged in #96 so there is now a conflict

**Comment by tremes** [comment]
@lholmquist Yeah sure.

**Pull request merged**